common/tier4_autoware_utils/include/tier4_autoware_utils/ros/published_time_publisher.hpp

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TIER4_AUTOWARE_UTILS__ROS__PUBLISHED_TIME_PUBLISHER_HPP_
+#define TIER4_AUTOWARE_UTILS__ROS__PUBLISHED_TIME_PUBLISHER_HPP_
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <autoware_internal_msgs/msg/published_time.hpp>
+#include <std_msgs/msg/header.hpp>
+
+#include <functional>
+#include <map>
+#include <string>
+
+namespace tier4_autoware_utils
+{
+using autoware_internal_msgs::msg::PublishedTime;
+
+struct GidHash
+{
+  size_t operator()(const rmw_gid_t & gid) const noexcept
+  {
+    // Hashing function that computes a hash value for the GID
+    std::hash<std::string> hasher;
+    return hasher(std::string(reinterpret_cast<const char *>(gid.data), RMW_GID_STORAGE_SIZE));
+  }
+};
+
+struct GidEqual
+{
+  bool operator()(const rmw_gid_t & lhs, const rmw_gid_t & rhs) const noexcept
+  {
+    return std::memcmp(lhs.data, rhs.data, RMW_GID_STORAGE_SIZE) == 0;
+  }
+};
+
+class PublishedTimePublisher
+{
+public:
+  static std::unique_ptr<PublishedTimePublisher> create(
+    rclcpp::Node * node, const std::string & end_name = "/debug/published_time",
+    const rclcpp::QoS & qos = rclcpp::QoS(1))
+  {
+    const bool use_published_time = node->declare_parameter<bool>("use_published_time", false);
+    if (use_published_time) {
+      return std::unique_ptr<PublishedTimePublisher>(
+        new PublishedTimePublisher(node, end_name, qos));
+    } else {
+      return nullptr;
+    }
```

Comment:
We don't need a way to disable this with declared params, you can remove this `use_published_time` param and its logic.

Could you remove this create function and make it a constuctor instead?

Reply:
Makes sense, I removed.